### ISSUE TITLE
New collection created with every bot interaction

### ISSUE THREAD

**Issue by pchest** [brucepro/Memoir#27]
I am running your extension on Pop!_OS 22.04. Most features seem to work fine, but I noticed that with every interaction with the bot that a new collection is created. It isn't clear to me whether this is referring to the creation of a new data entry or the collection itself is being recreated with each interaction. If it's the latter, then I wouldn't expect there to be any persistent memory across sessions.

![image](https://github.com/brucepro/Memoir/assets/23111423/be29a40f-74c5-4635-a22f-6d7c18a14b2c)

Additionally, there appears to be no measurable growth in the size of stored memories in qdrant over time, which further suggests that it's the collection that's recreated and that there is no persistent memory.

![image](https://github.com/brucepro/Memoir/assets/23111423/9f25209f-9893-40be-90a2-bcb605ef3b27)

I'll be happy to run tests to see if persistent memory is in operation, or if it is not, to help resolve this issue.

Regardless, thank you for your time and effort on this project.

**Comment by pchest** [comment]
@brucepro I tried changing the name to remove the space, but I'm having the same issue. 

![image](https://github.com/brucepro/Memoir/assets/23111423/c2e4b0f6-068c-4776-8f86-3b082e6cb2e0)

I look forward to seeing the new version!

**Comment by pchest** [comment]
Yes, I am able to view the dashboard and verified that it contains the bot entry.

![image](https://github.com/brucepro/Memoir/assets/23111423/de6a02df-5907-46f1-aed8-29e91064f88a)

**Comment by pchest** [comment]
Got it. Thanks for looking into this. Let me know if you need any configuration information.

**Comment by v1ckxy** [comment]
@brucepro I modified "create_vector_db_if_missing" class just to countermeasure this (plus a few more things), please have a look:
https://github.com/brucepro/Memoir/commit/ebb20c39c08936ae2bf8760c01e7d89105fe78c4#diff-fe5b88e7d82b4e3af15a7050de06f0b0cd055eebce5c0227226911816418ad72

qdrant_client has the property collection_exists just to check if the collection already exists (returns true/false)
Looks like they added that API call in january.

Regards!

**Comment by pchest** [comment]
Thank you for looking into this! I'll test this out later today and close the issue if it works.

**Comment by v1ckxy** [comment]
TBH it was driving me nuts every time I saw the 409 _(Conflict)_ in qdrant server log 🤣

**Comment by pchest** [comment]
Well, fixing it sounds like a win-win! lol
